### PR TITLE
ULTRA Add west-south traffic in cross intersections

### PR DESCRIPTION
**Pull request by JenishPatel99**
To expose the ego vehicle to oncoming cars once it has completed the left turn and it is going westbound.

More details to come.

Side tasks:
- [x] Add traffic density based levels to taskX

### CODE REVIEW

File: ultra/ultra/config.yaml [hunk 1]
```diff
@@ -26,8 +26,30 @@ tasks:
...
+      test:  "scenarios/task1/train_no-traffic*"
+    easy:
+      train: "scenarios/task1/train_easy*"
+      test:  "scenarios/task1/train_easy*"
+    medium:
+      train: "scenarios/task1/train_medium*"
+      test:  "scenarios/task1/train_medium*"
+    hard:
+      train: "scenarios/task1/train_hard*"
+      test:  "scenarios/task1/train_hard*"
```
Comment (christianjans) [inline ultra/ultra/config.yaml:52]: What is Task 0? I don't see it in our `scenarios/` directory. The paths seem to refer to Task 1 scenarios? Also, should these paths include the task in the scenario names?

File: ultra/ultra/config.yaml [hunk 2]
```diff
@@ -26,8 +26,30 @@ tasks:
...
+      test:  "scenarios/taskX/test_intersection-specific*"
+    low-density:
+      train: "scenarios/taskX/train_low-density*"
+      train: "scenarios/taskX/test_low-density*"
+    mid-density:
+      train: "scenarios/taskX/train_mid-density*"
+      train: "scenarios/taskX/test_mid-density*"
+    high-density:
+      train: "scenarios/taskX/train_high-density*"
+      train: "scenarios/taskX/test_high-density*"
```
Comment (christianjans) [inline ultra/ultra/config.yaml:39]: Should these paths include `taskX` in the scenario names?

File: ultra/ultra/scenarios/common/distributions.py [hunk 3]
```diff
@@ -775,7 +775,24 @@
...
+                    "params": {
+                        "vehicle_cluster_size": (1, 1),
+                        "time_between_cluster": (20, 30),
+                        "time_for_each_cluster": 1,
+                    },
+                },
+                "has_turn": True,
+                "deadlock_optimization": True,
+            },
             "east-west": None,
```
Comment (christianjans) [inline ultra/ultra/scenarios/common/distributions.py:796]: Do you know if it's still possible for a continuous control baseline to learn with these extra social vehicles?

File: ultra/ultra/config.yaml (hunk 1 above)
Comment (JenishPatel99) [reply ultra/ultra/config.yaml:52]: I do not know myself hahaha, I thought yesterday I had fixed that issue, but it seems that I was working on the same branch on two different repos, and I believe the changes were not correctly implemented. This also reflects on the comment below (task info in the scenario name). I have committed the new changes but I will push the changes once I have done a bit more training on the cross intersection

File: ultra/ultra/scenarios/common/distributions.py (hunk 3 above)
Comment (JenishPatel99) [reply ultra/ultra/scenarios/common/distributions.py:796]: I had some preliminary training done with sac, but it was not much, however I will run a bit longer experiment with the older reward model and then with the new one. I will convert this PR back to a draft and once completed I will let you know 👍

**Comment by JenishPatel99** [comment]
Changes proposed in PR are outdated and some are not required. Closing this PR

**Pull request closed without merging**